q2_assembly/tests/test_helpers.py

```diff
@@ -57,6 +60,55 @@ def test_is_valid_shortuuid(self):
         self.assertTrue(self.is_valid_shortuuid(true_shortuuid))
         self.assertFalse(self.is_valid_shortuuid(false_shortuuid))
 
+    def test_collate_genomes_dnafastaformat_single(self):
+        genomes1 = DNAFASTAFormat(
+            self.get_data_path("dna-fasta-format/dna-sequences1.fasta"), "r"
+        )
+        collated_genomes = collate_genomes(genomes=[genomes1])
+        self.assertEqual(len(os.listdir(collated_genomes.path)), 2)
+        exp_files = ["ref1.fasta", "ref2.fasta"]
+        actual_files = sorted(os.listdir(collated_genomes.path))
+        self.assertEqual(actual_files, exp_files)
+
+    def test_collate_genomes_dnafastaformat_multiple(self):
+        genomes1 = DNAFASTAFormat(
+            self.get_data_path("dna-fasta-format/dna-sequences1.fasta"), "r"
+        )
+        genomes2 = DNAFASTAFormat(
+            self.get_data_path("dna-fasta-format/dna-sequences2.fasta"), "r"
+        )
+        collated_genomes = collate_genomes(genomes=[genomes1, genomes2])
+        exp_files = ["ref1.fasta", "ref2.fasta", "ref5.fasta", "ref6.fasta"]
+        actual_files = sorted(os.listdir(collated_genomes.path))
+        self.assertEqual(len(os.listdir(collated_genomes.path)), 4)
+        self.assertEqual(exp_files, actual_files)
```

Comment:
I like that you added those new checks but that's not exactly what I meant. First of all, this new check is redundant with the old one - you can remove the assertion on the length of the list as you are explicitly checking file names. Second, what I meant you should add is rather a check that the _content_ of the files is correct in case where you take a FeatureData[Sequence] artifact and split it the file within it into GenomeData[DNASequence].